plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizard.java

```diff
@@ -0,0 +1,131 @@
+package org.jboss.tools.openshift.internal.ui.wizard.deployimage;
+
+import static org.jboss.tools.common.ui.WizardUtils.runInWizard;
+
+import java.lang.reflect.InvocationTargetException;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.jobs.IJobChangeEvent;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.jobs.JobChangeAdapter;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchWizard;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.jboss.tools.common.ui.JobUtils;
+import org.jboss.tools.openshift.internal.common.core.UsageStats;
+import org.jboss.tools.openshift.internal.common.core.job.JobChainBuilder;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+import org.jboss.tools.openshift.internal.ui.dialog.ResourceSummaryDialog;
+import org.jboss.tools.openshift.internal.ui.job.DeployImageJob;
+import org.jboss.tools.openshift.internal.ui.job.RefreshResourcesJob;
+import org.jboss.tools.openshift.internal.ui.wizard.common.ResourceLabelsPage;
+
+/**
+ * The deploy image wizard that allows you to deploy a docker image to openshift
+ * with a reasonable set of openshift resources
+ * OpenShift template
+ * 
+ * @author jeff.cantrill
+ */
+public class DeployImageWizard extends Wizard implements IWorkbenchWizard {
+
+	private static final String TITLE = "Deploy Image to OpenShift";
+
+	private static final String OPENSHIFT_EXPLORER_VIEW_ID = "org.jboss.tools.openshift.express.ui.explorer.expressConsoleView";
+
+	private IDeployImageParameters model;
+
+	public DeployImageWizard(IDeployImageParameters wizardModel) {
+		setWindowTitle(TITLE);
+		setNeedsProgressMonitor(true);
+		this.model = wizardModel;
+	}
+
+	@Override
+	public void addPages() {
+		addPage(new DeployImagePage(this, model));
+		addPage(new DeploymentConfigPage(this, model));
+		addPage(new ServicesAndRoutingPage(this, model));
+		addPage(new ResourceLabelsPage(this, model));
+	}
+
+	@Override
+	public boolean performFinish() {
+		final DeployImageJob deployJob = new DeployImageJob(model);
+		deployJob.addJobChangeListener(new JobChangeAdapter(){
+
+			@Override
+			public void done(IJobChangeEvent event) {
+				IStatus status = event.getResult();
+				if(JobUtils.isOk(status) || JobUtils.isWarning(status)) {
+					Display.getDefault().syncExec(new Runnable() {
+						@Override
+						public void run() {
+							final String message = NLS.bind(
+									"Results of deploying image \"{0}\".", model.getName());
+							new ResourceSummaryDialog(
+									getShell(), 
+									deployJob.getResources(),
+									TITLE,
+									message).open();
+						}
+					});
+					Display.getDefault().asyncExec(new Runnable() {
```

Comment:
we should prolly refactor that part to a utility class, instead of duplicating that code